### ISSUE TITLE
CANNOT IMPORT 'SIMILARITIES '

### ISSUE THREAD

**Issue by williamwei2** [NicolasHug/Surprise#336]
Hi, I feel stuck in the installing process of using surprise pakage under Colab.google.research framewrok, the details is going like this:

1. copy the repository

INPUT:
!git clone https://github.com/NicolasHug/surprise.git
OUTPUT:
Cloning into 'surprise'...
remote: Enumerating objects: 36, done.
remote: Counting objects: 100% (36/36), done.
remote: Compressing objects: 100% (30/30), done.
remote: Total 4257 (delta 10), reused 14 (delta 6), pack-reused 4221
Receiving objects: 100% (4257/4257), 6.36 MiB | 5.03 MiB/s, done.
Resolving deltas: 100% (2964/2964), done.

2. locate the corresondig file
INPUT:
cd surprise
OUTPUT:
/content/surprise

3.Install
INPUT
!python setup.py install
OUTPUT
(TOO LONG , SO IGNORE DESCRIPITION ABOUT ADDING FILES)

4.import package
INPUT:
import pandas as pd
from surprise import Reader
from surprise import Dataset
from surprise.model_selection import cross_validate
from surprise import NormalPredictor
from surprise import KNNBasic
from surprise import KNNWithMeans
from surprise import KNNWithZScore
from surprise import KNNBaseline
from surprise import SVD
from surprise import BaselineOnly
from surprise import SVDpp
from surprise import NMF
from surprise import SlopeOne
from surprise import CoClustering
from surprise.accuracy import rmse
from surprise import accuracy
from surprise.model_selection import train_test_split

OUTPUT
ImportError Traceback (most recent call last)
in ()
1 import pandas as pd
----> 2 from surprise import Reader
3 from surprise import Dataset
4 from surprise.model_selection import cross_validate
5 from surprise import NormalPredictor

2 frames
/content/surprise/surprise/prediction_algorithms/algo_base.py in ()
8 unicode_literals)
9
---> 10 from .. import similarities as sims
11 from .predictions import PredictionImpossible
12 from .predictions import Prediction

ImportError: cannot import name 'similarities'

ANYONE COULD HELP THIS ?

THANKS IN ADVANCE

**Comment by NicolasHug** [comment]
did something go wrong during the installation?

**Comment by williamwei2** [comment]
Yup. It leaves an 'cannot import error' about the similarities class when I try to import the surprise package in colabsearch.google environment, a Linux environment.



---Original---

**Comment by williamwei2** [comment]
nothing goes wrong during installation



---Original---

**Comment by NicolasHug** [comment]
have you properly followed the install from source instructions?
If yes, please paste the entire log with installation details

**Comment by williamwei2** [comment]
I think so. Could you please verify this. The log file is attached.



------------------&nbsp;原始邮件&nbsp;------------------
发件人:&nbsp;"Nicolas Hug"<notifications@github.com&gt;;
发送时间:&nbsp;2020年2月13日(星期四) 晚上10:09
收件人:&nbsp;"NicolasHug/Surprise"<Surprise@noreply.github.com&gt;;
抄送:&nbsp;"691260559"<691260559@qq.com&gt;;"Author"<author@noreply.github.com&gt;;
主题:&nbsp;Re: [NicolasHug/Surprise] CANNOT IMPORT 'SIMILARITIES ' (#336)




have you properly followed the install from source instructions?
 If yes, please paste the entire log with installation details

**Comment by NicolasHug** [comment]
You probably attached it as email attachment. paste it on github instead pleae

**Comment by williamwei2** [comment]
[suprise_log.txt](https://github.com/NicolasHug/Surprise/files/4198912/suprise_log.txt)

**Comment by NicolasHug** [comment]
indeed it seems like it goes normally. Sorry I'm not sure I can help here. I think there are past discussions about using colab, you might have better luck there.

It's not possible to use conda or pip from within colab?

**Comment by williamwei2** [comment]
Yeah， it could use pip or !pip, however, an unknown error would be incurred, as  is followed:
pip setup.py install
ERROR: unknown command "setup.py"
![unkown error](https://user-images.githubusercontent.com/6024662/74496551-f1ed7500-4f15-11ea-90cc-a1af68d1d6a3.png)

**Comment by williamwei2** [comment]
Using Python instead of pip would be working to next line

**Comment by williamwei2** [comment]
Anyway , thx a lot, i will try to find colab topic anywhere

**Comment by Om-patel2398** [comment]
I am facing the same issue please help. 
While working in colab I tried the example

'''''''''''''''''''''''
from surprise import SVD
from surprise import Dataset
from surprise.model_selection import cross_validate

data = Dataset.load_builtin('ml-100k')

algo = SVD()

cross_validate(algo, data, measures=['RMSE', 'MAE'], cv=5, verbose=True)
'''''''''''''''''''''''

it gave error

'''''''''''''''''''
EOFError                                  Traceback (most recent call last)
<ipython-input-3-8ce051c7fe0e> in <module>()
      1 from surprise import Dataset
      2 
----> 3 Dataset.load_builtin('ml-100k')

/usr/local/lib/python3.6/dist-packages/surprise/dataset.py in load_builtin(cls, name, prompt)
     87                 print('Dataset ' + name + ' could not be found. Do you want '
     88                       'to download it? [Y/n] ', end='')
---> 89                 choice = input().lower()
     90 
     91                 if choice in ['yes', 'y', '', 'omg this is so nice of you!!']:

EOFError: EOF when reading a line
'''''''''''''''''''

So, in short, I cannot work anywhere on surprise

**Comment by Om-patel2398** [comment]
I fixed the above error by changing input() to input('')
